### PR TITLE
[no-release-notes] projection mappings perf

### PR DESCRIPTION
**Pull request by max-hoffman**

### CODE REVIEW

**Comment by max-hoffman** [comment]
#newbenchmark

**Comment by max-hoffman** [comment]
#newbenchmark

File: go/libraries/doltcore/sqle/history_table.go [hunk 1]
```diff
@@ -41,6 +42,12 @@ const (
 
 	// CommitDateCol is the name of the column containing the commit date in the result set
 	CommitDateCol = "commit_date"
+
+	// history tags implement new projection interface, they should not correspond to real tags
+	// and are private to the history table
+	commitHashTag uint64 = iota
+	committerTag
+	commitDateTag
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/history_table.go:50]: we have tags reserved tags for this history table here: https://github.com/dolthub/dolt/blob/efdf426a664b235f53704ffd80a982ab6c11d535/go/libraries/doltcore/schema/reserved_tags.go#L42

File: go/libraries/doltcore/sqle/history_table.go [hunk 2]
```diff
@@ -208,14 +216,54 @@ func transformFilters(ctx *sql.Context, filters ...sql.Expression) []sql.Express
...
 }
 
-func (ht HistoryTable) WithProjections(colNames []string) sql.Table {
-	projectedTable := ht.doltTable.WithProjections(colNames)
-	ht.doltTable = projectedTable.(*DoltTable)
-	return &ht
+func (ht *HistoryTable) WithProjections(colNames []string) sql.Table {
+	nt := *ht
+	nt.projectedCols = make([]uint64, len(colNames))
+	// todo(max): is updating the child table necessary?
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/history_table.go:222]: I'm pretty sure the answer it yes

File: go/libraries/doltcore/sqle/integration_test/history_table_test.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ import (
 )
 
 func TestHistoryTable(t *testing.T) {
-	SkipByDefaultInCI(t)
+	//SkipByDefaultInCI(t)
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/integration_test/history_table_test.go:34]: reskip

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

**Pull request merged**